src/ol/renderer/layer.js

```diff
@@ -152,8 +152,10 @@ ol.renderer.Layer.prototype.scheduleExpireCache = function(frameState, tileSourc
      */
     var postRenderFunction = function(tileSource, map, frameState) {
       var tileSourceKey = ol.getUid(tileSource).toString();
-      tileSource.expireCache(frameState.viewState.projection,
-          frameState.usedTiles[tileSourceKey]);
+      if (frameState.usedTiles !== null && frameState.usedTiles[tileSourceKey] !== null) {
```

Comment:
`frameState.usedTiles` should never be null. And I think that `frameState.usedTiles[tileSourceKey]` would rather be `undefined` and not `null`. Does it still work if you change the `if` condition to `if (tileSourceKey in frameState.usedTiles)`?